tasks/component/index.js

```diff
@@ -0,0 +1,109 @@
+const {
+  ROOT_DIR,
+  createGenerator,
+  fs,
+  gulp,
+  inquirer,
+  interpolate,
+  mergeStream,
+  mkdirsSync,
+  pascalToKebab,
+  path,
+  pkgDir,
+  rename,
+} = require('../../utils');
+
+const defaults = {
+  srcRoot: 'src',
+  testRoot: 'test/unit',
+};
+
+const mapArgsToAnswers = ([name, srcPath = ''], { srcRoot = defaults.srcRoot }) => ({
+  name,
+  srcPath,
+  srcRoot,
+  moveon: true,
+});
+
+const prompts = [
+  {
+    name: 'name',
+    message: 'What is your component called?',
+  },
+  {
+    name: 'srcRoot',
+    message: 'Provide the path to the root of the project source code:',
+    default: defaults.srcRoot,
+  },
+  {
+    name: 'srcPath',
+    message: 'Provide the path to the directory where the component will be created:',
+  },
+  {
+    type: 'confirm',
+    name: 'moveon',
+    message: 'Continue?'
+  },
+];
+
+const callback = (answers, done) => {
+  if (!answers.moveon) {
+    return done();
+  }
+
+  // Handle name.
+  const sanitizedName = answers.name.replace(/\s+/g, '');
+  const slug = pascalToKebab(sanitizedName);
+  answers.slug = slug;
+  answers.sanitizedName = sanitizedName;
+
+  // Compute and validate src and test dirs.
+  const srcDir = path.join(ROOT_DIR, answers.srcRoot, answers.srcPath, slug);
+  const testDir = path.join(ROOT_DIR, defaults.testRoot, answers.srcPath, slug);
+
+  if (fs.existsSync(srcDir) || fs.existsSync(testDir)) {
+    done('Please ensure that the component-specific folder do not exist within the source or test directories');
```

Comment:
"folders do" or "folder does", but not both (probably the former?).

Reply:
Whoops, good catch. I believe this originally referred to a single folder.